scripts/extract_from_ISO.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+#
+# Copyright (C) 2019 onwards, Samveen S. Gulati
+#
+# Script to extract FL1 or FL2 files from Lenovo BIOS update ISO image
+#
+
+# EL TORITO offset count (thanks to Hamish Coleman)
+FAT_OFFSET=71680
+
+USAGE="${0} [[-1|--fl1] [-2|--fl2]] {[-i|--iso] ISO_FILE} {[-d|--desc] DESC_FILE} [[-m|--model] PREFIX] [[-s|--suffix] SUFFIX] | [-h|--help]
+    -1|--fl1    Extract BIOS update file (.FL1)
+    -2|--fl2    Extract EC update file (.FL2) (Default if no option)
+    -i|--iso    Update ISO to extract the file from
+    -d|--desc   Path of Descriptions.txt file
+    -m|--model  Optional model prefix to attach to extracted filenames.
+    -s|--suffix Optional suffix to attach to extracted filenames.
+    -h|--help   Show this help"
+
+# Get and parse options
+OPTIONS=$(getopt -o 12i:d:m:s:h --long fl1,fl2,iso:,desc:,model:,suffix:,help -n "$0" -- "$@")
+# Note the quotes around `$OPTIONS': they are essential!
+eval set -- "$OPTIONS"
+
+while true ; do
+    case "$1" in
+        -1|--fl1) FL1=Y ; shift ;;
+        -2|--fl2) FL2=Y ; shift ;;
+        -i|--iso) ISO=$2 ; shift 2 ;;
+        -d|--desc) DESC=$2 ; shift 2 ;;
+        -m|--model) MODEL="${2%%.}." ; shift 2 ;;
+        -s|--suffix) SUFFIX=".${2##.}" ; shift 2 ;;
+        -h|--help) echo "$USAGE"; exit ;;
+        --) shift ; break ;;
+        *) echo "Internal error!" ; exit 1 ;;
+    esac
+done
+
+# Reasonable Defaults: extract only FL2
+if [[ -z $FL1 ]] && [[ -z $FL2 ]]; then
+    FL2=Y
+fi
+
+[[ -n "$ISO" ]]  || { echo -e "Error: Missing ISO filename\n${USAGE}" >&2; exit 2; }
+[[ -f "$ISO" ]]  || { echo -e "Error: ${ISO} not found or not a file\n${USAGE}" >&2; exit 4; }
+[[ -f "$DESC" ]]  || { echo -e "Error: Missing Descriptions file.\n${USAGE}" >&2; exit 8; }
+[[ -n "$MODEL" ]] && echo "Using ${MODEL} as prefix" >&2 || echo "No model name provided. Not using any prefix" >&2
+[[ -n "$SUFFIX" ]] && echo "Using ${SUFFIX} as suffix" >&2 || echo "Not using any suffix." >&2
+
+# ./scripts/ISO_copyFL2 from_iso $< $@ $(1)
+
+if ! which mdir >/dev/null; then
+    echo "Fatal: mtools package not installed or not in path" >&2
+    exit 16
+fi
+
+read -r FL1VERNUM FL1VER FL2VERNUM FL2VER < <(grep -E "^${ISO%%.iso*}" ../Descriptions.txt| \
```

Comment:
Did you mean "$DESC" here, instead of ../Descriptions.txt ?

Reply:
> Did you mean "$DESC" here, instead of ../Descriptions.txt ?

Yeah. That's a typo. I'll fix it.
 
> Also, you are treating a free-form text string as a set of fixed format fields when you extract the various version numbers.

I looked at the format of the file, and I see that there is a bit of commonality between all the lines. I'm counting on this structure. The issue is with the EC versioning. there is no way that I know off to figure out the EC version number inside a given ISO. However, I'm hoping there is something, which I have not yet figured out myself, which allows that. Any suggestions?